geth/accounts_test.go

```diff
@@ -300,6 +311,9 @@ func TestRecoverAccount(t *testing.T) {
 	if !whisperService.HasIdentity(crypto.ToECDSAPub(common.FromHex(pubKeyCheck))) {
 		t.Errorf("identity not injected into whisper: %v", err)
 	}
+	if !swarmService.HasIdentity(crypto.ToECDSAPub(common.FromHex(pubKeyCheck))) {
```

Comment:
- no need to check if identity exist in Swarm
- all those identity management were just copied from Whisper (where they make sense, as you can have multiple identities decrypting incoming traffic, in swarm single protocol identity is enough, isn't it?)
- checking that swarm service runs ok might be beneficial though